tests/tests.js

```diff
@@ -0,0 +1,76 @@
+// reference to the V8's test cases
+// also see https://github.com/v8/v8/blob/master/test/mjsunit/array-reverse.js
+
+QUnit.test('test primitive literal', function(assert) {
+  assert.deepEqual([], [].reverse(), 'empty array');
+  assert.deepEqual([8, 6, 4, 2], [2, 4, 6, 8].reverse(), 'array of int');
+  assert.deepEqual(
+    [0.8, 0.6, 0.4],
+    [0.4, 0.6, 0.8].reverse(),
+    'array of double'
+  );
+  assert.deepEqual(
+    ['str4', 'str3', 'str2'],
+    ['str2', 'str3', 'str4'].reverse(),
+    'array of string'
+  );
+});
+
+QUnit.test('test with holes', function(assert) {
+  assert.deepEqual([4, 3, , 1], [1, , 3, 4].reverse());
+  assert.deepEqual([4, , 2, 1], [1, 2, , 4].reverse());
+  assert.deepEqual([5, , 3, , 1], [1, , 3, , 5].reverse());
+});
+
+QUnit.test('test with undefined', function(assert) {
```

Comment:
it seems the array includes `undefined` is not affected by the bug because `undefined` is not a real literal 🤣

Reply:
so maybe we can drop this test case?

Reply:
Sorry, I misunderstood what you mean. This test case is duplicated with the following, holes are treated like undefined elements